Continuous Integration workflow started to fail since Feb 6, 2025 with several mypy errors

Since Feb 6, 2025 `make mypy` started to fail as follows:

```
Collecting lxml-stubs
  Downloading lxml_stubs-0.5.1-py3-none-any.whl.metadata (1.6 kB)
Collecting types-PyYAML
  Using cached types_PyYAML-6.0.12.20241230-py3-none-any.whl.metadata (1.8 kB)
Collecting types-html5lib
  Using cached types_html5lib-1.1.11.20241018-py3-none-any.whl.metadata (1.9 kB)
Collecting types-requests
  Using cached types_requests-2.32.0.20241016-py3-none-any.whl.metadata (1.9 kB)
Requirement already satisfied: urllib3>=2 in /opt/hostedtoolcache/Python/3.9.21/x64/lib/python3.9/site-packages (from types-requests) (2.3.0)
Downloading lxml_stubs-0.5.1-py3-none-any.whl (13 kB)
Using cached types_PyYAML-6.0.12.20241230-py3-none-any.whl (20 kB)
Using cached types_html5lib-1.1.11.20241018-py3-none-any.whl (17 kB)
Using cached types_requests-2.32.0.20241016-py3-none-any.whl (15 kB)
Installing collected packages: lxml-stubs, types-requests, types-PyYAML, types-html5lib
Successfully installed lxml-stubs-0.5.1 types-PyYAML-6.0.12.20241230 types-html5lib-1.1.11.20241018 types-requests-2.32.0.20241016
src/pwncollege_cli/pwncollege_cli.py:271: error: Value of type "Union[PageElement, Tag, NavigableString]" is not indexable  [index]
src/pwncollege_cli/pwncollege_cli.py:271: error: Item "AttributeValueList" of "Union[Any, str, AttributeValueList]" has no attribute "removeprefix"  [union-attr]
src/pwncollege_cli/pwncollege_cli.py:271: error: Invalid index type "str" for "NavigableString"; expected type "Union[SupportsIndex, slice[Any, Any, Any]]"  [index]
src/pwncollege_cli/pwncollege_cli.py:272: error: Item "PageElement" of "Union[PageElement, Tag, NavigableString]" has no attribute "find"  [union-attr]
src/pwncollege_cli/pwncollege_cli.py:272: error: Unexpected keyword argument "class_" for "find" of "str"  [call-arg]
src/pwncollege_cli/pwncollege_cli.py:272: error: Item "None" of "Union[Any, PageElement, int, None]" has no attribute "text"  [union-attr]
src/pwncollege_cli/pwncollege_cli.py:272: error: Item "int" of "Union[Any, PageElement, int, None]" has no attribute "text"  [union-attr]
src/pwncollege_cli/pwncollege_cli.py:275: error: Item "PageElement" of "Union[PageElement, Tag, NavigableString]" has no attribute "find"  [union-attr]
src/pwncollege_cli/pwncollege_cli.py:275: error: Unexpected keyword argument "class_" for "find" of "str"  [call-arg]
src/pwncollege_cli/pwncollege_cli.py:275: error: Item "None" of "Union[Any, PageElement, int, None]" has no attribute "stripped_strings"  [union-attr]
Installing missing stub packages:
/opt/hostedtoolcache/Python/3.9.21/x64/bin/python -m pip install lxml-stubs types-PyYAML types-html5lib types-requests

Found 26 errors in 1 file (checked 3 source files)
src/pwncollege_cli/pwncollege_cli.py:275: error: Item "int" of "Union[Any, PageElement, int, None]" has no attribute "stripped_strings"  [union-attr]
src/pwncollege_cli/pwncollege_cli.py:317: error: Value of type "Union[PageElement, Tag, NavigableString]" is not indexable  [index]
src/pwncollege_cli/pwncollege_cli.py:317: error: Item "AttributeValueList" of "Union[Any, str, AttributeValueList]" has no attribute "removeprefix"  [union-attr]
src/pwncollege_cli/pwncollege_cli.py:317: error: Invalid index type "str" for "NavigableString"; expected type "Union[SupportsIndex, slice[Any, Any, Any]]"  [index]
src/pwncollege_cli/pwncollege_cli.py:318: error: Item "PageElement" of "Union[PageElement, Tag, NavigableString]" has no attribute "find"  [union-attr]
src/pwncollege_cli/pwncollege_cli.py:318: error: Unexpected keyword argument "class_" for "find" of "str"  [call-arg]
src/pwncollege_cli/pwncollege_cli.py:318: error: Item "None" of "Union[Any, PageElement, int, None]" has no attribute "text"  [union-attr]
src/pwncollege_cli/pwncollege_cli.py:318: error: Item "int" of "Union[Any, PageElement, int, None]" has no attribute "text"  [union-attr]
src/pwncollege_cli/pwncollege_cli.py:321: error: Item "PageElement" of "Union[PageElement, Tag, NavigableString]" has no attribute "find"  [union-attr]
src/pwncollege_cli/pwncollege_cli.py:321: error: Unexpected keyword argument "class_" for "find" of "str"  [call-arg]
src/pwncollege_cli/pwncollege_cli.py:321: error: Item "None" of "Union[Any, PageElement, int, None]" has no attribute "stripped_strings"  [union-attr]
src/pwncollege_cli/pwncollege_cli.py:321: error: Item "int" of "Union[Any, PageElement, int, None]" has no attribute "stripped_strings"  [union-attr]
src/pwncollege_cli/pwncollege_cli.py:365: error: "PageElement" has no attribute "find"  [attr-defined]
src/pwncollege_cli/pwncollege_cli.py:366: error: "PageElement" has no attribute "find"  [attr-defined]
src/pwncollege_cli/pwncollege_cli.py:367: error: "PageElement" has no attribute "find"  [attr-defined]
src/pwncollege_cli/pwncollege_cli.py:370: error: "PageElement" has no attribute "find"  [attr-defined]
make: *** [Makefile:21: mypy-install-types] Error 1
```

Investigate what it is going on.

--- a/src/pwncollege_cli/pwncollege_cli.py
+++ b/src/pwncollege_cli/pwncollege_cli.py
@@ -268,11 +268,22 @@ class PwnCollegeCLI:
         dojos = []
         b = bs4.BeautifulSoup(response.content, "html.parser")
         for e in b.find_all("a", href=re.compile(r"/dojo/")):
+            assert isinstance(e, bs4.element.Tag)
+            assert isinstance(e["href"], str)
+
             dojo_id = e["href"].removeprefix("/dojo/")
-            dojo_name = e.find(class_="card-title").text
+
+            card_title = e.find(class_="card-title")
+            if not card_title:
+                break
+            dojo_name = card_title.text
+
+            card_text = e.find(class_="card-text")
+            if not card_text:
+                break
 
             hacking = 0  # noone could be hacking on dojo
-            for s in e.find(class_="card-text").stripped_strings:
+            for s in card_text.stripped_strings:
                 if "Hacking" in s:
                     hacking = int(s.split()[0])
                     continue
@@ -314,11 +325,22 @@ class PwnCollegeCLI:
         modules = []
         b = bs4.BeautifulSoup(response.content, "html.parser")
         for e in b.find_all("a", href=re.compile(f"^/{dojo}/[a-z0-9-]+/?$")):
+            assert isinstance(e, bs4.element.Tag)
+            assert isinstance(e["href"], str)
+
             module_id = e["href"].removeprefix(f"/{dojo}/").removesuffix("/")
-            module_name = e.find(class_="card-title").text
+
+            card_title = e.find(class_="card-title")
+            if not card_title:
+                break
+            module_name = card_title.text
+
+            card_text = e.find(class_="card-text")
+            if not card_text:
+                break
 
             hacking = 0  # noone could be hacking on module
-            for s in e.find(class_="card-text").stripped_strings:
+            for s in card_text.stripped_strings:
                 if "Hacking" in s:
                     hacking = int(s.split()[0])
                     continue
@@ -362,14 +384,27 @@ class PwnCollegeCLI:
             b.find_all("div", id=re.compile("challenges-header")),
             b.find_all("div", id=re.compile("challenges-body")),
         ):
-            challenge_id = body.find("input", id="challenge-id")["value"]
-            challenge_name = body.find("input", id="challenge")["value"]
-            challenge_title = header.find(
-                "h4", class_="challenge-name"
-            ).text.strip()
-            challenge_description = body.find(
-                "div", class_="embed-responsive"
-            ).text.strip()
+            assert isinstance(body, bs4.element.Tag)
+            assert isinstance(header, bs4.element.Tag)
+
+            t = body.find("input", id="challenge-id")
+            assert isinstance(t, bs4.element.Tag)
+            challenge_id = t["value"]
+            assert isinstance(challenge_id, str)
+
+            t = body.find("input", id="challenge")
+            assert isinstance(t, bs4.element.Tag)
+            challenge_name = t["value"]
+            assert isinstance(challenge_name, str)
+
+            t = header.find("h4", class_="challenge-name")
+            assert isinstance(t, bs4.element.Tag)
+            challenge_title = t.text.strip()
+
+            t = body.find("div", class_="embed-responsive")
+            assert isinstance(t, bs4.element.Tag)
+            challenge_description = t.text.strip()
+
             challenges.append(
                 Challenge(
                     id=challenge_id,

